codecov.yml

```diff
@@ -0,0 +1,26 @@
+# https://docs.codecov.com/docs/
+# https://docs.codecov.com/docs/codecov-yaml
+
+codecov:
+  branch: main # Set the default branch
+
+parsers:
+  go:
+    partials_as_hits: true
+
+ignore:
+  # Ignore our end-to-end test code
+  - "go/test/endtoend"
+  - "go/*/endtoend"
+  - "go/vt/*/endtoend"
+  - "go/cmd/vttestserver" # This relies on end-to-end test packages
```

Comment:
Should we exclude all `cached_size.go` files as well as generated code?

Reply:
Added here: https://github.com/vitessio/vitess/pull/14967/commits/56f139f94695bc757dff392b081dd21587da235b

Please let me know if you think we should remove any of the added ignore lines. Thanks!